staging/src/k8s.io/client-go/plugin/pkg/client/auth/gcp/gcp.go

```diff
@@ -113,7 +113,16 @@ type gcpAuthProvider struct {
 	persister   restclient.AuthProviderConfigPersister
 }
 
+var warnOnce sync.Once
+
 func newGCPAuthProvider(_ string, gcpConfig map[string]string, persister restclient.AuthProviderConfigPersister) (restclient.AuthProvider, error) {
+	// deprecated in v1.22, remove in v1.25
+	// this should be updated to use klog.Warningf in v1.24 to more actively warn consumers
```

Comment:
Is there going to be some kind of process in place to ensure this happens? Otherwise is there anything programatic we could do to query the version at runtime and update the error? 

Also it seems slightly risky to explicitly communicate the exact version it is being disabled.  What if there is a lot of pushback at depreciation and its replacement and we need a few extra release cycles to hammer out those concerns?

Reply:
> Is there going to be some kind of process in place to ensure this happens?

I'd expect a milestoned issue

> Also it seems slightly risky to explicitly communicate the exact version it is being disabled. What if there is a lot of pushback at depreciation and its replacement and we need a few extra release cycles to hammer out those concerns?

This won't merge until the client-go exec plugin GA merges. That is the target for any issues or concerns. Once that is released as GA, setting a definite removal version for the other auth plugins is reasonable.

Reply:
Opened #103498 to track removal in 1.25.  #102890 is tracking GA of client-go credential plugins.